functions/AuthUtils.js

```diff
@@ -0,0 +1,186 @@
+const { Issuer } = require('openid-client');
+const { generators } = require('openid-client');
+const jwt = require('jsonwebtoken');
+const cookie = require('cookie');
+
+const getOpenIDClient = async () => {
+    const issuer = await Issuer.discover(`https://${process.env.AUTH0_DOMAIN}`);
+    return new issuer.Client({
+        client_id: process.env.AUTH0_CLIENT_ID,
+        redirect_uris: [`${process.env.URL}/.netlify/functions/callback`],
+        response_types: ['id_token'],
+    });
+};
+
+const generateNetlifyJWT = async (tokenData) => {
+    const iat = Math.floor(Date.now() / 1000);
+    const twoWeeksInSeconds = 14 * 24 * 3600;
+    const exp = Math.floor(iat + twoWeeksInSeconds);
+    //copy over appropriate properties from the original token data
+    //Refer to Netlify Documentation for token formatting - https://docs.netlify.com/visitor-access/role-based-access-control/#external-providers
+    const netlifyTokenData = {
+        exp,
+        iat,
+        updated_at: iat,
+        aud: tokenData.aud,
+        sub: tokenData.sub,
+        app_metadata: {
+            authorization: {
+                roles: tokenData[`${process.env.AUTH0_TOKEN_NAMESPACE}/roles`],
+            },
+        },
+    };
+    return await jwt.sign(netlifyTokenData, process.env.TOKEN_SECRET);
+};
+
+const generateAuth0LoginCookie = (nonce, encodedStateStr) => {
+    const cookieData = { nonce, state: encodedStateStr };
+    return cookie.serialize('auth0_login_cookie', JSON.stringify(cookieData), {
+        secure: !process.env.NETLIFY_DEV === 'true',
+        path: '/',
+        maxAge: 30 * 60 * 1000,
+        httpOnly: true,
+    });
+};
+
+const generateEncodedStateString = (route) => {
+    const state = { route: route || '/', nonce: generators.nonce() };
+    const stateBuffer = Buffer.from(JSON.stringify(state));
+    return stateBuffer.toString('base64');
+};
+
+const generateAuth0LoginResetCookie = () => {
+    return cookie.serialize('auth0_login_cookie', 'Auth0 Login Cookie Reset', {
```

Comment:
What's the second param ("Auth0 Login Cookie Reset") for?

Reply:
The actual body of the cookie. Doesn't need to have anything I guess. Maybe empty string?

Reply:
Removed and using empty string now.